src/allmydata/test/test_node.py

```diff
@@ -26,10 +26,14 @@ class TestNode(Node):
     PORTNUMFILE='DEFAULT_PORTNUMFILE_BLANK'
 
 class TestCase(testutil.SignalMixin, unittest.TestCase):
+
+    @defer.inlineCallbacks
     def setUp(self):
         testutil.SignalMixin.setUp(self)
         self.parent = LoggingMultiService()
         self.parent.startService()
+        self._available_port = yield iputil.allocate_tcp_port()
```

Comment:
Changes in this module look unrelated, just general reliability improvements for the test suite by avoiding random port collisions?  If so it might be nice to split this off into a separate pr.